ls1intum/Artemis
# [Exam] Code-Editor Bugfixes

### Checklist


### Motivation and Context
Notes from Test Session:
- [x] On Startup, the `Submission`'s `isSynced` flag was set to `false`. We use `true` at the start of the conduction. 
- [x] Worker Issue: Refused to create a worker from '...' because it violates the following Content Security Policy directive...
![image](https://user-images.githubusercontent.com/15110960/87410531-bccd7780-c5c6-11ea-847a-601fdd3ae7e1.png)
- [x] Results of a submission were loaded into the wrong exercise. _Possible fix in:_ 162965666e3ddbb8d25d720e15d46fdb05792b3d
- [x] When you push your changes locally and afterwards press "refresh files" in the online code editor, the wrong exercise files are loaded _Possible fix in:_ 162965666e3ddbb8d25d720e15d46fdb05792b3d
- [x] TypeError: `this.fileSession[this.selectedFile]` is undefined (refresh files)
![image](https://user-images.githubusercontent.com/15110960/87410695-edadac80-c5c6-11ea-8d45-2bd806c970f3.png)



### Description
1. Set the `isSynced` to `true` and `submitted` to `false` for the initial `ProgrammingSubmission`.
   Track the submitted status using the `onCommitStateChange` event.
2. Add additional guards against `undefined`, see point 5 above.
3. Fix navigation issue from 162965666e3ddbb8d25d720e15d46fdb05792b3d solved point 3 & 4 above.
4. Every component now has its own service, thereby effectively eliminating the potential overlaps.

### Steps for Testing
1. Log in to Artemis
2. Create an exam with multiple programming exercises. 
3. Open the console & check the network tab.
4. Interact with the exam, switch between exercises, create submissions, edit files etc. 
5. All exercises should load correctly, and the correct id should be displayed in the console. 
6. No errors should be seen in the console & network tab

### Screenshots
